src/taipy/core/taipy.py

```diff
@@ -497,3 +498,52 @@ def export_scenario(
         _ScenarioManagerFactory._build_manager()._export(scenario_id, folder_path)
     for job_id in entity_ids.job_ids:
         _JobManagerFactory._build_manager()._export(job_id, folder_path)
+
+
+def get_parents(
+    entity: Union[TaskId, DataNodeId, PipelineId, Task, DataNode, Pipeline], parent_dict=None
+) -> Dict[str, Set[_Entity]]:
+    """Get the parents of an entity from itself or its identifier.
+
+    Parameters:
+        entity (Union[TaskId^, DataNodeId^, PipelineId^, Task, DataNode, Pipeline]): The entity or its identifier
+            to get the parents.<br/>
+    Returns:
+        Dict[str, Set[_Entity]]: The dictionary of parents matching the corresponding entity with the key
+        is the level of the parent for example 'scenarios', 'pipelines', 'tasks' and the value is a list
+        of the parent entities. An empty dictionary if the entity does not have parents.
```

Comment:
I find it very complex and difficult to understand. 
Tell me what you think about this proposal.
```suggestion
        Dict[str, Set[_Entity]]: The dictionary of all parent entities.
            They are grouped by their type (`Scenario^`, `Pipelines^`, or `tasks^`) so each key
            corresponds to a level of the parents and is among the following list: 
            'scenarios', 'pipelines', or 'tasks'. 
            An empty dictionary is returned if the entity does not have parents.<br/>
            Example: The following instruction returns all the pipelines that include the
            datanode identified by "my_datanode_id".
            `taipy.get_parents("id_of_my_datanode")["pipelines"]`
```

Reply:
@jrobinAV 
I think it's better! But we didn't include the set of parent entities as the value of the dictionary, how about `They are grouped by their type (`Scenario^`, `Pipelines^`, or `tasks^`) so each key corresponds to a level of the parents and the value is a set of the parent entities.`

Reply:
Yes that s better.